tests/53groups/03local.pl

```diff
@@ -0,0 +1,209 @@
+test "Add local group users",
+   requires => [ local_admin_fixture( with_events => 0 ), local_user_fixture( with_events => 0 ) ],
+
+   do => sub {
+      my ( $creator, $user ) = @_;
+
+      my $group_id;
+
+      matrix_create_group( $creator )
+      ->then( sub {
+         ( $group_id ) = @_;
+
+         matrix_invite_group_user( $creator, $group_id, $user );
+      })->then( sub {
+         matrix_accept_group_invite( $group_id, $user );
+      })->then( sub {
+         matrix_get_joined_groups( $user );
+      })->then( sub {
+         my ( $body ) = @_;
+
+         assert_json_keys( $body, qw( groups ) );
+         assert_json_list( my $group_ids = $body->{groups} );
```

Comment:
so that it doesn't get lost:

you've got a lot of duplicated code here. If you have to unpick pagination responses, that's even more reason for it to be factored out imho.

Reply:
The boiler plate for subsequent calls is `assert_deeply_eq( $body->{groups}, [ $group_id ] );`, which feels acceptable. Otherwise we may as well create a function that wraps `invite/accept/get_joined`